eventconsumer/src/main/scala/com/gu/notifications/events/Router.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.notifications.events
+
+import java.util.concurrent.TimeUnit
+
+import com.gu.notifications.events.dynamo.DynamoReportUpdater
+import com.gu.notifications.events.model.{AggregationCounts, NotificationReportEvent, S3ResultCounts}
+import com.gu.notifications.events.s3.{S3Event, S3EventProcessor}
+import com.gu.notifications.events.sqs.SqsEventReader
+import org.apache.logging.log4j.{LogManager, Logger}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.Try
+
+class Router(eventConsumer: S3EventProcessor, reportUpdater: DynamoReportUpdater)(implicit executionContext: ExecutionContext) {
+  private val logger: Logger = LogManager.getLogger(classOf[Router])
+
+  def sqsEventRoute(inputString: String): Unit = {
+    val s3Events: Seq[S3Event] = SqsEventReader.readSqsEventString(inputString)
+    val s3EventAggregateProviders = s3Events.map(s3Event => () => s3EventRoute(s3Event))
+    val eventualTriedNotificationCounts = callProvidersInSeries(s3EventAggregateProviders)
```

Comment:
reading `callProvidersInSeries` I don't see why there's any Future involved here as the processing is called in series, and the futures are all Future.successful.

If you want to trigger the computation in parallel I'd recommend having something like that
``` scala
val aggregates = s3Events.map(s3Event => () => s3EventRoute(s3Event))
Future.sequence(aggregates)
```

Reply:
As discussed, not a priority, doesn't require to be fixed right now